src/components/furniture/Space.vue

```diff
@@ -1,71 +1,29 @@
 <template>
-  <div class="separation" :style="cssStyle" @click="addHardwareToSpace">
-    <hardware-handle :space="space" :shelf="shelf"></hardware-handle>
+  <div class="separation" :style="{ width: this.mySpace.width + '%' }">
+    <div v-if="$route.path === '/add-hardware'" class="space">
```

Comment:
A component should work anywhere, without caring about its context.
Think about a Microwave or a Playstation. The playstation receives 3 props: electricity, the video cable for the TV, and the controllers for how many players. There is nothing INSIDE the playstation that will check for it's context. You can play in Bangladesh, Kuala Lumpur, etc. 

I think here you could easily receive the information by props to decide if this div should show instead of checking for a app-wide contexte